### DISCUSSION TITLE
Cannot run PointPillars Object Detection using Open3D-ML library

### DISCUSSION THREAD

**Discussion by btn6364** [community/community#112236] [Other Feature Feedback, Questions, & Ideas]
### Description

Hi, I'm running into an issue when running PointPillars pre-trained model using Open3D-ML. I followed the instruction on their Github repository https://github.com/isl-org/Open3D-ML/tree/main. I was able to get the visualization part running. But when I got to the 3D Object Detection using PointPillars, I ran into an issue `RuntimeError: IntArrayRef contains an int that cannot be represented as a SymInt: -4833712541423952424`. 

Here's the full log of the bug.
```
Traceback (most recent call last):
  File "/home/btn6364/VIPS/main/object_detection.py", line 32, in <module>
    result = pipeline.run_inference(data)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/btn6364/miniconda3/envs/vips/lib/python3.11/site-packages/open3d/_ml3d/torch/pipelines/object_detection.py", line 72, in run_inference
    results = model(data)
              ^^^^^^^^^^^
  File "/home/btn6364/miniconda3/envs/vips/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/btn6364/miniconda3/envs/vips/lib/python3.11/site-packages/open3d/_ml3d/torch/models/point_pillars.py", line 132, in forward
    x = self.extract_feats(inputs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/btn6364/miniconda3/envs/vips/lib/python3.11/site-packages/open3d/_ml3d/torch/models/point_pillars.py", line 104, in extract_feats
    voxels, num_points, coors = self.voxelize(points)
                                ^^^^^^^^^^^^^^^^^^^^^
  File "/home/btn6364/miniconda3/envs/vips/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/btn6364/miniconda3/envs/vips/lib/python3.11/site-packages/open3d/_ml3d/torch/models/point_pillars.py", line 117, in voxelize
    res_voxels, res_coors, res_num_points = self.voxel_layer(res)
                                            ^^^^^^^^^^^^^^^^^^^^^
  File "/home/btn6364/miniconda3/envs/vips/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/btn6364/miniconda3/envs/vips/lib/python3.11/site-packages/open3d/_ml3d/torch/models/point_pillars.py", line 354, in forward
    ans = voxelize(points,
          ^^^^^^^^^^^^^^^^
  File "/home/btn6364/miniconda3/envs/vips/lib/python3.11/site-packages/open3d/ml/torch/python/ops.py", line 274, in voxelize
    *_torch.ops.open3d.voxelize(points=points,
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/btn6364/miniconda3/envs/vips/lib/python3.11/site-packages/torch/_ops.py", line 502, in __call__
    return self._op(*args, **kwargs or {})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: IntArrayRef contains an int that cannot be represented as a SymInt: -4833712541423952424
```

Here's the code that I have:
```
import os
import open3d.ml as _ml3d
import open3d.ml.torch as ml3d
import open3d as o3d
o3d.visualization.webrtc_server.enable_webrtc()

cfg_file = "../Open3D-ML/ml3d/configs/pointpillars_kitti.yml"
cfg = _ml3d.utils.Config.load_from_file(cfg_file)

model = ml3d.models.PointPillars(**cfg.model)
cfg.dataset['dataset_path'] = "../datasets/Kitti"
dataset = ml3d.datasets.KITTI(cfg.dataset.pop('dataset_path', None), **cfg.dataset)
pipeline = ml3d.pipelines.ObjectDetection(model, dataset=dataset, device="gpu", **cfg.pipeline)

# download the weights.
ckpt_folder = "./logs/"
os.makedirs(ckpt_folder, exist_ok=True)
ckpt_path = ckpt_folder + "pointpillars_kitti_202012221652utc.pth"
pointpillar_url = "https://storage.googleapis.com/open3d-releases/model-zoo/pointpillars_kitti_202012221652utc.pth"
if not os.path.exists(ckpt_path):
    cmd = "wget {} -O {}".format(pointpillar_url, ckpt_path)
    os.system(cmd)

# load the parameters.
pipeline.load_ckpt(ckpt_path=ckpt_path)

test_split = dataset.get_split("test")
data = test_split.get_data(0)

# run inference on a single example.
# returns dict with 'predict_labels' and 'predict_scores'.
result = pipeline.run_inference(data)

# evaluate performance on the test set; this will write logs to './logs'.
pipeline.run_test()
```

I already printed the `data` variable to the console, which worked. But when I passed it into `pipeline.run_inference()`, it didn't work. I'm not sure why this happen. Any hint to solve this problem will be helpful. Thanks!

By the way, I'm running this on: 
- Pytorch: 2.0.1+cu117. 
- CUDA: 12.4.

**Comment by kuciapakrystian** [comment]
The error message you're encountering indicates an issue with integer representation, possibly due to an overflow or underflow problem. This could be happening at any point in the processing pipeline, but it's most likely related to the voxelization step.

Here are a few steps you can take to diagnose and potentially resolve the issue:

Check Input Data: Ensure that your input data is properly formatted and within the expected range. Sometimes issues can arise if the input data contains outliers or extreme values.

Check Voxelization Parameters: Review the parameters used for voxelization, such as voxel size, grid size, and coordinate system. Ensure that these parameters are appropriate for your data and that they are not causing numerical instability.

Check Torch Version Compatibility: Make sure that the versions of PyTorch and other related libraries (such as Open3D-ML) are compatible with each other. It's possible that a compatibility issue could be causing unexpected behavior.

Debugging: Try to isolate the exact point in the code where the error occurs. You can do this by adding print statements or using a debugger to inspect the values of variables at different points in the pipeline.

Check for Updates: Check for any updates or bug fixes in the Open3D-ML repository or related dependencies. It's possible that the issue you're encountering has already been addressed in a newer version.

Community Support: Reach out to the developers or community forums for Open3D-ML or PyTorch for assistance. They may have encountered similar issues and can provide guidance or solutions.

Consider Numerical Stability: If the issue persists, consider whether there are any numerical stability issues in the voxelization or other parts of the pipeline. This could include using smaller numerical types or applying normalization techniques to the input data.

By following these steps and investigating the possible causes of the error, you should be able to diagnose and potentially resolve the issue with running PointPillars using Open3D-ML.

**Comment by Haris-AAI** [comment]
@btn6364 Hey! Were you able to resolve this issue? I am getting same error while trying to run inference on single frame. I tried to take only 1000 points for debugging purposes but still got this error.

**Reply by shahriar0651** [reply]
I am also getting the same error. @btn6364 @Haris-AAI, did you find any way to fix this issue?
